enterprise/views.py

```diff
@@ -686,19 +803,26 @@ def get(self, request, enterprise_uuid, course_id):
             * No enterprise customer uuid kwarg `enterprise_uuid` in request.
             * No enterprise customer found against the enterprise customer
                 uuid `enterprise_uuid` in the request kwargs.
+            * No course is found in database against the provided `course_id`.
         """
         # Verify that all necessary resources are present
         verify_edx_resources()
 
         enterprise_customer, course, modes = self.get_base_details(enterprise_uuid, course_id)
 
+        # Create a link between the user and the enterprise customer if it
+        # does not already exist.
+        enterprise_customer_user, __ = EnterpriseCustomerUser.objects.get_or_create(
```

Comment:
@haikuginger I am doing this on both `GET` and `POST` (guard purpose) requests. According to the discussion on [ENT-379|https://openedx.atlassian.net/browse/ENT-379] we should link the learner to enterprise customer before showing the enterprise landing page.